src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
-// write your code here
+const url
+= 'https://mate-academy.github.io/phone-catalogue-static/api/phones.json';
+
+const controller = new AbortController();
+const timeoutId = setTimeout(() => controller.abort(), 5000);
+
+const getPhones
+  = fetch(url, { signal: controller.signal })
+    .then((response) => response.json())
+    .then((data) => {
+      return data;
+    });
+
+clearTimeout(timeoutId);
+
+let html = '';
+
+function getPhonesDetails() {
+  try {
+    getPhones.then((data) => {
+      data.forEach(x => {
+        const htmlSegment
+        = `<div class="phoneName">
+          <p>${x.name}</p>
+        </div>`;
+
+        html += htmlSegment;
```

Comment:
it is not critical, but I'd rather use `append` method instead of straightforward concatenation